govcd/extension_test.go

```diff
@@ -28,3 +30,82 @@ func (vcd *TestVCD) Test_GetExternalNetwork(check *C) {
 	check.Assert(externalNetwork.Name, Equals, networkName)
 	check.Assert(externalNetwork.Type, Equals, expectedType)
 }
+
+func (vcd *TestVCD) Test_CreateExternalNetwork(check *C) {
+	if vcd.skipAdminTests {
+		check.Skip("Configuration org != 'Sysyem'")
+	}
+	networkName := vcd.config.VCD.ExternalNetwork
+	if networkName == "" {
+		check.Skip("No external network provided")
+	}
+
+	externalNetworkRef, err := GetExternalNetworkByName(vcd.client, networkName)
+	check.Assert(err, IsNil)
+	if *externalNetworkRef != (types.ExternalNetworkReference{}) {
+		externalNetwork := NewExternalNetwork(&vcd.client.Client)
+		externalNetwork.ExternalNetwork = &types.ExternalNetwork{
+			HREF: externalNetworkRef.HREF,
+		}
+		err = externalNetwork.DeleteWait()
+		check.Assert(err, IsNil)
+	}
+
+	results, err := vcd.client.QueryWithNotEncodedParams(nil, map[string]string{
```

Comment:
Recommend to extract to specific function. We have similar for queering media, vappTemplate

Reply:
I wasn't quite sure where to put it, but it is in `extension.go` now.